src/parser/toplevel.lisp

```diff
@@ -432,14 +446,9 @@
   (instances       (util:required 'instances)       :type toplevel-define-instance-list :read-only nil)
   (specializations (util:required 'specializations) :type toplevel-specialize-list      :read-only nil))
 
-;;
-;; Empty package for reading (package) forms
-;;
-(defpackage #:coalton-impl/parser/read
-  (:use))
-
-(defun read-program (stream file &key (mode (error "you must supply this bozo!")))
-  "Read a PROGRAM from the SE:FILE."
+(defun read-program (stream file &key mode)
+  "Read a PROGRAM from FILE (an instance of source-error:file).
+If MODE is :file, a package form is required."
```

Comment:
Document all acceptable `mode`s here and their meanings.

Reply:
done